clang/lib/CIR/Dialect/Transforms/SCFPrepare.cpp

```diff
@@ -147,6 +147,31 @@ struct hoistLoopInvariantInCondBlock : public OpRewritePattern<ForOp> {
     return true;
   }
 
+  // Return true for loop invariant operation and push it to initOps.
+  bool isLoopInvariantOp(Operation *op, ForOp forOp,
+                         SmallVector<Operation *> &initOps) const {
+    if (!op)
+      return false;
+    if (isa<ConstantOp>(op) || isLoopInvariantLoad(op, forOp)) {
+      initOps.push_back(op);
+      return true;
+    } else if (isa<BinOp>(op) &&
+               isLoopInvariantOp(op->getOperand(0).getDefiningOp(), forOp,
```

Comment:
Are these operands guaranteed not to be block arguments? If there's a chance they are, this will segfault while trying to get a defining op!

Reply:
For block arguments, I assume getDefiningOp() will return null and isLoopInvariantOp() will return false by line 153 before reference.

Reply:
Gotcha, can you add one testcase that cover such examples?

Reply:
Yes, sure. will do.